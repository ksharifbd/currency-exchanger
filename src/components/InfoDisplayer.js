import React from 'react';
import PropTypes from 'prop-types';
import Box from '@material-ui/core/Box';
import styled from '@material-ui/core/styles/styled';

const labelStyles = {
  color: 'rgba(0, 0, 0, 0.54)',
  padding: 0,
  fontSize: '1rem',
  lineHeight: 1,
  letterSpacing: '0.00938em',
};

const InfoDisplayerWrapper = styled(Box)({
  width: '13.75rem',
});
const InfoDisplayerLabel = styled(Box)({
  ...labelStyles,
  marginBottom: '.25rem',
});
const InfoDisplayerSymbol = styled(Box)({
  ...labelStyles,
  display: 'inline-block',
  marginRight: '.25rem',
});

const InfoDisplayer = ({ label, symbol, amount, hasWrapper }) => {
  const content = (
    <>
      <InfoDisplayerLabel component="p">{label}</InfoDisplayerLabel>
      <Box>
        <InfoDisplayerSymbol component="span">{symbol}</InfoDisplayerSymbol>
        <Box component="span">{amount}</Box>
      </Box>
    </>
  );

  return (
    <>
      {hasWrapper ? (
        <InfoDisplayerWrapper id="infoDisplayerWrapper">
          {content}
        </InfoDisplayerWrapper>
      ) : (
        content
      )}
    </>
  );
};

InfoDisplayer.propTypes = {
  label: PropTypes.string.isRequired,
  symbol: PropTypes.string.isRequired,
  amount: PropTypes.string.isRequired,
  hasWrapper: PropTypes.bool,
};

InfoDisplayer.defaultProps = {
  hasWrapper: false,
};

export default InfoDisplayer;
