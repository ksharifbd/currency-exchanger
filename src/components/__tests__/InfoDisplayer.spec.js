import React from 'react';
import { createMount } from '@material-ui/core/test-utils';
import InfoDisplayer from '../InfoDisplayer';

describe('Info Displayer', () => {
  let mount;

  beforeAll(() => {
    mount = createMount();
  });

  afterAll(() => {
    mount.cleanUp();
  });

  const tree = (hasWrapper = false) =>
    mount(
      <InfoDisplayer
        label="label"
        symbol="$"
        amount="10"
        hasWrapper={hasWrapper}
      />
    );

  it('should render the intial setup correctly', () => {
    expect(tree()).toMatchSnapshot();
  });

  it('should render the setup with wrapper correctly', () => {
    expect(tree(true)).toMatchSnapshot();
  });

  it('should render wrapper correctly', () => {
    expect(tree(true).exists('#infoDisplayerWrapper')).toEqual(true);
  });
});
