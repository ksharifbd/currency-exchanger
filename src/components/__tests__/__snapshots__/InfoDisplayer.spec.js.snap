// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info Displayer should render the intial setup correctly 1`] = `
<InfoDisplayer
  amount="10"
  hasWrapper={false}
  label="label"
  symbol="$"
>
  <Styled(Styled(MuiBox))
    component="p"
  >
    <p
      className="Styled(MuiBox)-root-1"
    >
      label
    </p>
  </Styled(Styled(MuiBox))>
  <Styled(MuiBox)>
    <div
      className="MuiBox-root MuiBox-root-3"
    >
      <Styled(Styled(MuiBox))
        component="span"
      >
        <span
          className="Styled(MuiBox)-root-4"
        >
          $
        </span>
      </Styled(Styled(MuiBox))>
      <Styled(MuiBox)
        component="span"
      >
        <span
          className="MuiBox-root MuiBox-root-5"
        >
          10
        </span>
      </Styled(MuiBox)>
    </div>
  </Styled(MuiBox)>
</InfoDisplayer>
`;

exports[`Info Displayer should render the setup with wrapper correctly 1`] = `
<InfoDisplayer
  amount="10"
  hasWrapper={true}
  label="label"
  symbol="$"
>
  <Styled(Styled(MuiBox))
    id="infoDisplayerWrapper"
  >
    <Styled(MuiBox)
      className="Styled(MuiBox)-root-6"
      id="infoDisplayerWrapper"
    >
      <div
        className="MuiBox-root MuiBox-root-8 Styled(MuiBox)-root-6"
        id="infoDisplayerWrapper"
      >
        <Styled(Styled(MuiBox))
          component="p"
        >
          <p
            className="Styled(MuiBox)-root-9"
          >
            label
          </p>
        </Styled(Styled(MuiBox))>
        <Styled(MuiBox)>
          <div
            className="MuiBox-root MuiBox-root-10"
          >
            <Styled(Styled(MuiBox))
              component="span"
            >
              <span
                className="Styled(MuiBox)-root-11"
              >
                $
              </span>
            </Styled(Styled(MuiBox))>
            <Styled(MuiBox)
              component="span"
            >
              <span
                className="MuiBox-root MuiBox-root-12"
              >
                10
              </span>
            </Styled(MuiBox)>
          </div>
        </Styled(MuiBox)>
      </div>
    </Styled(MuiBox)>
  </Styled(Styled(MuiBox))>
</InfoDisplayer>
`;
